src/components/GraphiQL.js

```diff
@@ -478,9 +478,14 @@ export class GraphiQL extends React.Component {
     let jsonVariables = null;
 
     try {
-      jsonVariables = JSON.parse(variables);
+      jsonVariables =
+        variables && variables.trim() !== '' ? JSON.parse(variables) : null;
     } catch (error) {
-      jsonVariables = null;
+      throw new Error(`Variables are invalid JSON: ${error.message}.`);
```

Comment:
Instead of showing a string, should we create a JSON error format for these errors, like the ones that server gives back?
```js
{
  "errors": [
    {
      "message": "Variables are invalid JSON."
    }
  ]
}
```